client/jetpack-cloud/sections/pricing/jpcom-masterbar/index.tsx

```diff
@@ -88,42 +92,225 @@ const JetpackComMasterbar: React.FC< Props > = ( { pathname } ) => {
+
+			if ( ! menu ) {
+				return;
+			}
+
+			const toggleSubmenu = function () {
+				toggleMenuItem( btn, menu );
+			};
+
+			menu.addEventListener( 'click', function ( e ) {
+				// If user clicks menu backdrop
+				if ( e.target === menu ) {
+					toggleSubmenu();
+				}
+			} );
+
+			btn.addEventListener( 'click', function ( e ) {
+				e.preventDefault();
+
+				if ( btn.getAttribute( 'aria-expanded' ) === 'false' ) {
+					collapseExpandedMenu();
+				}
+
+				toggleSubmenu();
+			} );
+
+			const backBtn = menu.querySelector( '.js-menu-back' );
 
-		if ( btn.getAttribute( 'aria-expanded' ) === 'false' ) {
-			collapseExpandedMenu();
+			if ( backBtn ) {
+				backBtn.addEventListener( 'click', function () {
+					toggleSubmenu();
+				} );
+			}
+
+			// Collapse menu when focusing out
+			const lastFocusable = getLastFocusableElement( menu );
+
+			if ( lastFocusable ) {
+				lastFocusable.addEventListener( 'focusout', toggleSubmenu );
+			}
 		}
 
-		toggleMenuItem( btn, menu );
-	};
+		// Close expanded menu on Esc keypress
+		function onKeyDown( e: KeyboardEvent ) {
+			if ( e.key === 'Escape' ) {
+				collapseExpandedMenu();
+			}
+		}
+		// END SUBMENU TOGGLE FUNCTIONALITY
+
+		// USER MENU FUNCTIONALITY
+		const userMenu = document.querySelector( '.js-user-menu' ) as HTMLDivElement;
+		const userBtn = document.querySelector( '.js-user-menu-btn' ) as HTMLAnchorElement;
+
+		function onDocumentClick( { target }: MouseEvent ) {
+			if ( ! userMenu?.contains( target as Node ) ) {
+				userBtn?.setAttribute( 'aria-expanded', 'false' );
+				userMenu.hidden = true;
+			}
+		}
+
+		function toggleUserMenu() {
+			const expanded = userBtn.getAttribute( 'aria-expanded' ) === 'true' || false;
+
+			if ( expanded ) {
+				document.removeEventListener( 'click', onDocumentClick );
+			} else {
+				document.addEventListener( 'click', onDocumentClick );
+			}
+
+			userBtn.setAttribute( 'aria-expanded', String( ! expanded ) );
+			userMenu.hidden = ! userMenu.hidden;
+		}
+
+		function onUserBtnClick( e: MouseEvent ) {
+			e.preventDefault();
+			e.stopPropagation();
+
+			toggleUserMenu();
+		}
+		// END USER MENU FUNCTIONALITY
+
+		// MOBILE MENU FUNCTIONALITY
+		const MOBILE_BP = 900; // lrg-screen breakpoint
+		const mobileMenu = document.querySelector( '.js-mobile-menu' ) as HTMLDivElement;
+		const mobileBtn = document.querySelector( '.js-mobile-btn' );
+		const body = document.querySelector( 'body' );
+
+		function mobileMenuToggle() {
+			const expanded = mobileBtn?.getAttribute( 'aria-expanded' ) === 'true' || false;
+
+			mobileBtn?.setAttribute( 'aria-expanded', String( ! expanded ) );
+
+			if ( ! expanded ) {
+				mobileMenu?.classList.add( 'is-expanded' );
+				body?.classList.add( 'no-scroll' );
+			} else {
+				mobileMenu?.classList.remove( 'is-expanded' );
+				body?.classList.remove( 'no-scroll' );
+			}
+		}
 
-	useSubmenuBtn();
-	useUserMenu();
-	useMobileBtn();
```

Comment:
What would happen if you called these functions inside your new `useEffect` statement? 
Not sure if React would throw an error since these functions also employ `useEffect` 🤔 

That could allow for some of the code to stay separate/ more readable for now?

Reply:
Yeah I tried that initially, but React doesn’t allow useEffect to be called within any sort of callback. This is the specific error I get
`React Hook "useEffect" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.`